### DISCUSSION TITLE
HELP:  How to read Class->Field->Value

### DISCUSSION THREAD

**Discussion by uragan1987** [0xd4d/dnlib#535] [General]
hi guys

try to read Field Value 

i got looped module.Types, found my Type, 
got my Field from type.Fields.

but cant find there my string Value.

Can somebody help me here?

**Comment by wtfsck** [comment]
See FieldDef.InitialValue, assuming it isn't null it's just the raw bytes of the value. You have to check the field type to see what kind of data is stored there.

**Comment by uragan1987** [comment]
FieldDef.InitialValue is always null
but i have a value on dnspy

![initValNULL](https://github.com/0xd4d/dnlib/assets/13498010/331d75bc-4b0a-45cc-aa5c-36fb1e87f102)
![statefalse](https://github.com/0xd4d/dnlib/assets/13498010/e7194eb9-847a-4b90-b230-1c9c1f7de283)

or do i use it wrong?
